pkg/generate/config/field.go

```diff
@@ -267,6 +267,31 @@ type LateInitializeConfig struct {
 	MaxBackoffSeconds int `json:"max_backoff_seconds"`
 }
 
+// ReferencesConfig contains the instructions for how to add the referenced resource
+// configuration for a field.
+// Example:
+// ```
+// Integration:
+//    fields:
+//      ApiId:
+//        references:
+//          resource: API
+//          path: Status.APIID
+//```
+// The above configuration will result in generation of a new field 'APIIDRef'
+// of type 'AWSResourceReference' for ApiGatewayv2-Integration crd.
+// When 'APIIDRef' field is present in custom resource manifest, reconciler will
+// read the referred 'API' resource and copy the value from 'Status.APIID' in
+// 'Integration' resource's 'APIID' field
+type ReferencesConfig struct {
+	// Resource mentions the K8s resource which is read to resolve the
+	// reference
+	Resource string `json:"resource"`
+	// Path refers to the the path of field which should be copied
+	// to resolve the reference
+	Path string `json:"path"`
```

Comment:
In Crossplane, we run into cases where the referenced field needed to be parsed, hence we know have the concept of [extractor functions](https://github.com/crossplane/crossplane-runtime/blob/af4e148/pkg/reference/reference.go#L93) that you can write and point to in the [code generator](https://github.com/crossplane/crossplane-tools#reference-resolvers) to use if the [default one](https://github.com/crossplane/crossplane-runtime/blob/af4e148/pkg/reference/reference.go#L97) doesn't work for that case. I don't know if that'd be necessary for ACK but would definitely be helpful for Crossplane 🙂

Reply:
:+1: great idea. we can certainly add that functionality in a future enhancement

Reply:
Sounds good. FWIW, we'll definitely use this struct in Crossplane pipeline as well to automate the reference resolver generation 🙂  Great work!